src/sgame/sg_bot_ai.cpp

```diff
@@ -862,6 +892,47 @@ AINodeStatus_t BotActionFight( gentity_t *self, AIGenericNode_t *node )
 
 	// We have a valid visible target
 
+	if ( mind->hasOffmeshGoal )
+	{
+		// The target is visible, but not on the navmesh
+		ASSERT( G_Team( self ) == TEAM_HUMANS );
+		if ( !BotTargetIsVisible( self, self->botMind->goal, MASK_SHOT ) )
+		{
+			return STATUS_SUCCESS;
+		}
+		bool inRange = TargetInOffmeshAttackRange( self );
+		bool couldChangeToPrimary = BG_GetPlayerWeapon( &self->client->ps ) == WP_BLASTER && !WeaponIsEmpty( BG_PrimaryWeapon( self->client->ps.stats ), &self->client->ps );
+		if ( !inRange && BG_GetPlayerWeapon( &self->client->ps ) != WP_BLASTER )
+		{
+			G_ForceWeaponChange( self, WP_BLASTER );
```

Comment:
Is there some code to ensure the bot switches back to the primary weapon after finishing with the off-mesh target?

Reply:
There is now.